ci/sparklingWaterPipeline.groovy

```diff
@@ -382,4 +388,55 @@ def publishNightly() {
     }
 }
 
+def publishSparklingWaterDockerImage(String type, version, sparkMajorVersion) {
+    sh """
+        H2O_HOME=${env.WORKSPACE}/h2o-3 ./bin/build-kubernetes-images.sh ${type}
+        docker tag sparkling-water-${type}:${version} h2oai/sparkling-water-${type}:latest-nightly-${sparkMajorVersion}
+        docker push h2oai/sparkling-water-${type}:latest-nightly-${sparkMajorVersion}
+        docker rmi h2oai/sparkling-water-${type}:latest-nightly-${sparkMajorVersion}
+    """
+}
+
+def publishNightlyDockerImages() {
```

Comment:
This method seems to be very similar to https://github.com/h2oai/sparkling-water/pull/2161/files#diff-19c268ee1cd25cf26d1e8c79d4053ea0R388. Would be possible to share some logic?

Reply:
Good catch, done! 

I have however kept the method `publishDockerImage` in each file without sharing the logic as when I wrote it, there were too many conditions and switches between release & nightly. In that case, I think it is clearer to keep it separate